src/metabase/driver/ddl/mysql.clj

```diff
@@ -0,0 +1,142 @@
+  (let [{:keys [query params]} (qp/compile dataset-query)
+        db-spec (sql-jdbc.conn/db->pooled-connection-spec database)]
+    (jdbc/with-db-connection [conn db-spec]
+      (ddl.sql/execute! conn [(ddl.sql/drop-table-sql database (:table-name definition))])
+      ;; It is possible that this fails and rollback would not restore the table.
+      ;; That is ok, the persisted-info will be marked inactive and the next refresh will try again.
+      (execute-with-timeout! conn
+                             db-spec
+                             (.toMillis (t/minutes 10))
+                             (into [(ddl.sql/create-table-sql database definition query)] params))
+      {:state :success})))
+
+(defmethod ddl.i/unpersist! :mysql
+  [_driver database persisted-info]
+  (jdbc/with-db-connection [conn (sql-jdbc.conn/db->pooled-connection-spec database)]
+    (try
+      (ddl.sql/execute! conn [(ddl.sql/drop-table-sql database (:table_name persisted-info))])
+      (catch Exception e
+        (log/warn e)
+        (throw e)))))
+
+(defmethod ddl.i/check-can-persist :mysql
+  [database]
+  (let [schema-name (ddl.i/schema-name database (public-settings/site-uuid))
+        table-name (format "persistence_check_%s" (rand-int 10000))
+        db-spec (sql-jdbc.conn/db->pooled-connection-spec database)
+        steps [[:persist.check/create-schema
+                (fn check-schema [conn]
+                  (let [existing-schemas (->> ["select schema_name from information_schema.schemata"]
+                                              (ddl.sql/jdbc-query conn)
+                                              (map :schema_name)
+                                              (into #{}))]
+                    (or (contains? existing-schemas schema-name)
+                      (ddl.sql/execute! conn [(ddl.sql/create-schema-sql database)]))))
+                (fn undo-check-schema [conn]
+                  (ddl.sql/execute! conn [(ddl.sql/drop-schema-sql database)]))]
+               [:persist.check/create-table
+                (fn create-table [conn]
+                  (execute-with-timeout! conn
+                                         db-spec
+                                         (.toMillis (t/minutes 10))
+                                         [(ddl.sql/create-table-sql
+                                            database
+                                            {:table-name table-name
+                                             :field-definitions [{:field-name "field"
+                                                                  :base-type :type/Text}]}
+                                            "values (1)")]))
+                (fn undo-create-table [conn]
+                  (ddl.sql/execute! conn [(ddl.sql/drop-table-sql database table-name)]))]
+               [:persist.check/read-table
+                (fn read-table [conn]
+                  (ddl.sql/jdbc-query conn [(format "select * from %s.%s"
```

Comment:
Just curious, is there a reason you went with format and %s instead of

```
["select * from ?" (str schema-name "." "table-name")]
```

Reply:
Oh, that's just what postgres version did.